Sources/Basics/Cancellator.swift

```diff
@@ -0,0 +1,135 @@
+/*
+ This source file is part of the Swift.org open source project
+
+ Copyright (c) 2022 Apple Inc. and the Swift project authors
+ Licensed under Apache License v2.0 with Runtime Library Exception
+
+ See http://swift.org/LICENSE.txt for license information
+ See http://swift.org/CONTRIBUTORS.txt for Swift project authors
+ */
+
+import Dispatch
+import Foundation
+import TSCBasic
+
+public typealias CancellationHandler = (DispatchTime) throws -> Void
+
+public class Cancellator: Cancellable {
```

Comment:
Originally was named "terminator" which is cute but less accurate. Happy to take suggestions for a nicer name